tools/hangar/__snapshots__/test_corpus/valid/resource_captures.test.w_compile_tf-aws.md

```diff
@@ -338,7 +338,6 @@ class $Root extends $stdlib.std.Resource {
               $__arr__index_______if__index___0____index____arr_length__throw_new_Error__Index_out_of_bounds____return_arr_index______this_arrayOfStr__1_: ${$stdlib.core.liftObject(((arr, index) => { if (index < 0 || index >= arr.length) throw new Error("Index out of bounds"); return arr[index]; })(this.arrayOfStr, 1))},
               $__obj__key_______if____key_in_obj___throw_new_Error__Map_does_not_contain_key_____key______return_obj_key______this_mapOfNum___k1__: ${$stdlib.core.liftObject(((obj, key) => { if (!(key in obj)) throw new Error(`Map does not contain key: "${key}"`); return obj[key]; })(this.mapOfNum, "k1"))},
               $__obj__key_______if____key_in_obj___throw_new_Error__Map_does_not_contain_key_____key______return_obj_key______this_mapOfNum___k2__: ${$stdlib.core.liftObject(((obj, key) => { if (!(key in obj)) throw new Error(`Map does not contain key: "${key}"`); return obj[key]; })(this.mapOfNum, "k2"))},
-              $_this_myOptStr_______: ${$stdlib.core.liftObject((this.myOptStr ?? ""))},
```

Comment:
Why did the `?? ""` get removed here?

Reply:
Note that this:
```js
 $_this_myOptStr_______: ${$stdlib.core.liftObject((this.myOptStr ?? ""))},
```
was replaced by this:
```js
$this_myOptStr: ${$stdlib.core.liftObject(this.myOptStr)},
```
Previously the type_checker, pretty arbitrarily, evaluated `??` expressions to the phase of the left expression. So `??` expression this inflight code:
```wing
  pub inflight testCaptureOptional() {
    assert(this.myOptStr ?? "" == "myOptString");
  }
```
was evaluated to a preflight expression and therefore lifted the entire expression. This worked, but it's a bug, imagine this case:
```wing
  pub inflight testCaptureOptional() {
    let x = "";
    assert(this.myOptStr ?? x == "myOptString");
  }
```
Now the expression lifting will fail because there's an inflight variable inside of the preflight expression.
Our new code fixes this by correctly evaluating the binary `??` expression's phase. Since `""` is phase independent we the entire expression is phase independent can we only need to lift the inner `this.OptStr` part of it.